### DISCUSSION TITLE
When conky is clicked, conky will be displayed on top, how to make it always stay at the bottom.

### DISCUSSION THREAD

**Discussion by linux36524** [labwc/labwc#2318] [Q&A] [answered]
When conky is clicked, conky will be displayed on top, how to make it always stay at the bottom.
![1](https://github.com/user-attachments/assets/b33c097b-4309-49d8-b107-efbe79ec060e)

Same in labwc and wayfire

**Comment by Bluey26** [comment]
Hi, try with this in your `conky.conf` (~/.config/conky):

`own_window_type = 'dock',`

**Reply by linux36524** [reply]
> Hi, try with this in your `conky.conf` (~/.config/conky):
> 
> `own_window_type = 'dock',`

Seems useless
![20241108_19h05m58s_grim](https://github.com/user-attachments/assets/c2a8db05-1b08-43c0-87f1-ed183153516b)

**Reply by Bluey26** [reply]
If you already killed conky / it has updated, i suggest trying this code. It is what i am using at the moment, and the window is shown below:
```

conky.config = {
    out_to_wayland = true,
    alignment = 'top_right',
    background = true,
    border_width = 0,
    cpu_avg_samples = 2,
    default_color = 'white',
    default_outline_color = 'white',
    default_shade_color = 'white',
    double_buffer = true,
    draw_borders = false,
    draw_graph_borders = true,
    draw_outline = false,
    draw_shades = false,
    extra_newline = false,
    font = 'Hack:size=11',
    gap_x = 0,
    gap_y = 0,
    minimum_height = 5,
    minimum_width = 5,
    net_avg_samples = 2,
    no_buffers = true,
    out_to_console = false,
    out_to_ncurses = false,
    out_to_stderr = false,
    own_window = true,
    own_window_class = 'Conky',
    own_window_type = 'dock',
    own_window_colour = 'black',
    own_window_transparent = false,
    own_window_argb_visual = true,
    own_window_argb_value = 90,
    own_window_hints = 'undecorated,below,sticky,skip_taskbar,skip_pager',
    show_graph_range = false,
    show_graph_scale = false,
    stippled_borders = 0,
    update_interval = 1.0,
    uppercase = false,
    use_spacer = 'none',
    use_xft = true,
}
```

The relevant lines are:   own_window_hints , own_window and own_window_type , but i share it all because there may be something else.

**Reply by linux36524** [reply]
> If you already killed conky / it has updated, i suggest trying this code. It is what i am using at the moment, and the window is shown below:
> 
> ```
> 
> conky.config = {
>     out_to_wayland = true,
>     alignment = 'top_right',
>     background = true,
>     border_width = 0,
>     cpu_avg_samples = 2,
>     default_color = 'white',
>     default_outline_color = 'white',
>     default_shade_color = 'white',
>     double_buffer = true,
>     draw_borders = false,
>     draw_graph_borders = true,
>     draw_outline = false,
>     draw_shades = false,
>     extra_newline = false,
>     font = 'Hack:size=11',
>     gap_x = 0,
>     gap_y = 0,
>     minimum_height = 5,
>     minimum_width = 5,
>     net_avg_samples = 2,
>     no_buffers = true,
>     out_to_console = false,
>     out_to_ncurses = false,
>     out_to_stderr = false,
>     own_window = true,
>     own_window_class = 'Conky',
>     own_window_type = 'dock',
>     own_window_colour = 'black',
>     own_window_transparent = false,
>     own_window_argb_visual = true,
>     own_window_argb_value = 90,
>     own_window_hints = 'undecorated,below,sticky,skip_taskbar,skip_pager',
>     show_graph_range = false,
>     show_graph_scale = false,
>     stippled_borders = 0,
>     update_interval = 1.0,
>     uppercase = false,
>     use_spacer = 'none',
>     use_xft = true,
> }
> ```
> 
> The relevant lines are: own_window_hints , own_window and own_window_type , but i share it all because there may be something else.

Very strange, I used your configuration, but it still doesn't work. Thank you very much.
![20241108_19h30m12s_grim](https://github.com/user-attachments/assets/8c40a981-ce67-4f4a-b8cd-81ea560e1395)

**Reply by Bluey26** [reply]
By chance, do you use a panel as sfwbar or lxqt-panel? I am thinking that this may be related. I say this because in here i use lxqt-panel, and for some reason, the panel and conky seem to be "linked" somehow (killing the panel does hide conky in here).

**Reply by Bluey26** [reply]
I logged-in into a labwc session, without my LXQt DE. I have launched conky and then opened programs(i.e. browser). After this, conky window is shown below the browser (desired), and not as shown in your picture. My suspicion about the panel may be wrong, since this session has no panel and no deskto modules.

I am using `conky-git 1.21.7.r5.g71814776` but i have never experienced your issue with previous conky versions.

**Reply by linux36524** [reply]
I use waybar, but I turn off waybar, but the result is still the same.

**Comment by Consolatis** [comment]
The easiest solution would be using the wlr-layer-shell protocol and setting the layer to `bottom` but I don't know if conky supports that.

An alternative from the labwc side would be a [window rule](https://labwc.github.io/labwc-config.5.html#window_rules) combined with the [ToggleAlwaysOnBottom](https://labwc.github.io/labwc-actions.5.html#entry_action_name=togglealwaysonbottom) action like
```xml
<windowRules>
	<windowRule identifier="conky" serverDecoration="no" skipTaskbar="yes" skipWindowSwitcher="yes" ignoreFocusRequest="yes">
	    <action name="ToggleAlwaysOnBottom" />
	</windowRule>
</windowRules>
```
I've not tested the rule and I don't know if the `conky` identifier is correct.

**Reply by Bluey26** [reply]
The rule works. The proper identifier should be "Conky" (at least in my case, `own_window_class = 'Conky',`), but it seems to be case insensitive.

Since i already had `ToggleAlwaysOnBottom`(my conky panel is shown only in the desktop), i tried it with `ToggleAlwaysOnTop` and effectively conky is shown over all windows (testing purposes).

**Reply by linux36524** [reply]
> The rule works. The proper identifier should be "Conky" (at least in my case, `own_window_class = 'Conky',`), but it seems to be case insensitive.
> 
> Since i already had `ToggleAlwaysOnBottom`(my conky panel is shown only in the desktop), i tried it with `ToggleAlwaysOnTop` and effectively conky is shown over all windows (testing purposes).

![20241109_12h57m27s_grim](https://github.com/user-attachments/assets/caab4ebd-956b-4595-b527-3c2ba461ef33)
:(

**Reply by Consolatis** [reply]
Conky actually seems to support the layershell protocol from a quick look at its sources. So this looks like you don't need any window rule at all but just a way to tell conky to render in the "bottom" layer and ensure that conky is using the wayland backend.

**Reply by linux36524** [reply]
> Conky actually seems to support the layershell protocol from a quick look at its sources. So this looks like you don't need any window rule at all but just a way to tell conky to render in the "bottom" layer and ensure that conky is using the wayland backend.

But why does it lose its effect here? :(

**Reply by Consolatis** [reply]
Not sure, I don't use conky. Based on https://github.com/brndnmtthws/conky/wiki/Window-Configuration#wayland maybe the own_window config option is to blame and those 2 lines are all that is required.

**Reply by linux36524** [reply]
> Not sure, I don't use conky. Based on https://github.com/brndnmtthws/conky/wiki/Window-Configuration#wayland maybe the own_window config option is to blame and those 2 lines are all that is required.

out_to_x = false cannot open conky
Another problem is that you can move conky by holding alt + left mouse button

**Reply by Consolatis** [reply, marked as answer]
That sounds like your version of conky doesn't support the wayland backend and all issues mentioned are a result of that.

**Reply by Bluey26** [reply]
> > Not sure, I don't use conky. Based on https://github.com/brndnmtthws/conky/wiki/Window-Configuration#wayland maybe the own_window config option is to blame and those 2 lines are all that is required.
> 
> out_to_x = false cannot open conky Another problem is that you can move conky by holding alt + left mouse button

Did you find a way to solve the issue?

There was a discussion in the labwc IRC room that lead me to test some things, that shows that conky is having issues right now with its wayland support: https://github.com/brndnmtthws/conky/issues/1960 ; This was tested by me by compiling latest git master branch conky with the option to enable wayland support ( `-D BUILD_WAYLAND=ON`).

After that, i have added the corresponding lines to my `conky.conf`:

```
out_to_x = false,
out_to_wayland = true,
```

This time, conky had wayland support, but the window was not shown. This indicates that my previous conky version was running in X11 all the time, and that means that there is something that is missing in your case.

If you have not tested it already, i may suggest you to compile conky from the source, to test it. Also may be worth mentiong what OS are you using (from what i remember when i tested wayfire, it did not have issues with conky for me).
